src/modules/Users.pm

```diff
@@ -250,6 +254,29 @@ YaST::YCP::Import ("SSHAuthorizedKeys");
 ##-------------------------------------------------------------------------
 ##----------------- various routines --------------------------------------
 
+# Check whether the argument is a directory and is not empty.
+#
+# Return 1 (non-empty dir) or 0.
+#
+# Note: do NOT use FileUtils->GetSize() for this!
```

Comment:
I'm just curions, why not? What's the reason? Maybe it should be written here...

Reply:
Steffen has explained it:
..
2. the decision on whether to touch a home dir at all is based on whether the home dir is empty or not - but the implementation of the detection of this situation was wrong
..

The observed behavior differed between xfs (or ext4) and btrfs due to point 2. above. E.g. stat(DIR) returns 0 for empty btrfs directories while it is !=0 for empty xfs dirs.

Reply:
Oh, I see, so it is FS dependent... :open_mouth: